src/python/pants/backend/typescript/target_types.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from pants.core.goals.test import (
+    TestExtraEnvVarsField,
+    TestsBatchCompatibilityTagField,
+    TestTimeoutField,
+)
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    MultipleSourcesField,
+    OverridesField,
+    SingleSourceField,
+    Target,
+    TargetFilesGenerator,
+    generate_file_based_overrides_field_help_message,
+    generate_multiple_sources_field_help_message,
+)
+from pants.util.strutil import help_text
+
+TS_FILE_EXTENSIONS = (".ts", ".tsx")
```

Comment:
I doubt we're ready to deal with tsx at this stage? It invites a lot of questions of how we (wont/dont) handle the transpilation of these sources. I understand this is to enable the `test` goal, but I'm fairly sure running the compiler is what comes next? And in that case we'd want special targets for `.tsx` that it doesnt touch, at least not initally.